### ISSUE TITLE
[Accessibility Audit] Search: Notebooks

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34197]
### Use cases to replicate

Taken from the [user journey list document](https://docs.google.com/document/d/1IA90OFew5MTcxFndKS2Xn4vb8sOt68sj537P7AGj_Tk/edit?usp=sharing). Use this for further context.

Please take these following use cases and replicate them when running your audit.

- On the \*/notebooks page, I can:
  - Create a new Notebook
  - Import a Markdown file to create a new notebook
  - Search search the title and Markdown block content of Notebooks I created
  - Use tabs to toggle between displaying notebooks based on which ones I own, which ones I’ve starred, which ones my organization owns, and all the notebooks on my instances (Explore tab).
  - For every notebook on the list, I can see its name, description, owner, number of blocks, number of stars, and creation and update dates. I can click on a notebook to open it.
  - When creating or opening a notebook I have edit permissions to:
    - I can edit the title of the Notebook
    - I can delete the Notebook from the action menu
    - I can change the sharing permissions
    - I can add one or more of the following block types via the icon buttons, typing any string and selecting an available option from the dropdown that appears, or typing “/” and then type of block they want to create:
      - Markdown
        - I can use standard Markdown formatting such as lists, headings, and code formatting,
      - Sourcegraph search query
        - I can use the full Sourcegraph query language to generate search results as if I was in the main Sourcegraph app
      - File or code snippet
        - I can use the Sourcegraph query language to find a file of interest and optionally select a line range of interest
        - I can paste in a URL with a range of one or more lines
      - Symbol
        - I can use the Sourcegraph query language to search for symbols, optionally using the repo and file path filters to narrow my search
    - I can make a copy of the Notebook to My Notebooks
    - I can export the Notebook as a Markdown file
    - I can favorite the Notebook
    - I can run all runnable blocks in the Notebook by clicking the “Run all blocks” button
    - I can move blocks up or down in the list of blocks by clicking the “move up” or “move down” buttons (or Cmd/Ctrl + up arrow or Cmd/Ctrl + down arrow)
    - I can delete blocks by clicking the “Delete” button or Cmd/Ctrl + backspace
    - I can duplicate a block by clicking the “Duplicate” button or Cmd/Ctrl + D
    - I can save and render a block with Cmd/Ctrl + Return
    - I can edit a block by click the “Edit” button or pressing Return when a block is selected
    - I can open the full contents of any block results in a new tab by clicking the “Open in new tab” button
    - I can change the sharing settings of the Notebook to:
      - Share it with an organization of which I’m a member, which allows any member of the organization to edit it
      - Share it with an organization of which I’m a member and make it viewable to anyone on the instance, but editable only by members of my organization
      - Share it with my entire instance, but only I can edit
      - Only I can view or edit the notebook
  - When opening a notebook I only have read-access to:
    - I can view all blocks in the notebook and run searches
    - I can export and copy the notebook
    - I cannot change any of the block contents, block order, or metadata about the notebook

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by novoselrok** [comment]
I think we handled all of the stories, here are the issues I submitted:

* https://github.com/sourcegraph/sourcegraph/issues/35180
* https://github.com/sourcegraph/sourcegraph/issues/35179

And here are all (?) of the issues Marisa submitted:

* https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+author%3Amarisak+label%3Aaccessibility
